test/vol.c

```diff
@@ -1780,7 +1781,7 @@ exercise_reg_opt_oper(hid_t fake_vol_id, hid_t reg_opt_vol_id, H5VL_subclass_t s
     /* Push a new API context on the stack */
     /* (Necessary for the named datatype construction routines) */
     if (H5VL_SUBCLS_DATATYPE == subcls)
-        H5CX_push();
+        H5CX_push(&api_ctx);
```

Comment:
It looks like this context might reuse values from the one pushed at `vol.c:1728`

Reply:
True, but as I understand the block of memory isn't modified by `H5CX_pop()` (beyond potentially setting a few context return values), it's just removed from the context stack. Any modified values would persist.

Reply:
Yes, I agree with you, I'll memset it back to zeros for this (unusual) case.